.changeset/little-ads-return.md

```diff
@@ -0,0 +1,70 @@
+---
+"@makeswift/runtime": minor
+---
+
+BREAKING: The latest upgrade to the Makeswift client `getPages` method
+introduces sorting, path filtering, and pagination. This method was not
+previously paginated - in order to get all your pages, you may now use our
+`toArray` pagination helper method, which will automatically paginate through
+all results and aggregate them into an array:
+
+```tsx
+import { client } from '@/makeswift/client'
+import { MakeswiftPageResult } from '@makeswift/runtime/next'
+
+async function getAllPages(): Promise<MakeswiftPageResult[]> {
+  return await client.getPages().toArray()
+}
+```
+
+This change also deprecates the client `getSitemap` method, with the
+recommendation that sitemaps should now be generated using data returned from
+`getPages`. Note that the deprecation of `getSitemap` now involves the host
+being responsible for the construction of page paths in the sitemap (either with
+domain or path based localization). Below is an example that uses path-based
+localization with the `next-sitemap` library:
+
+```tsx pages/sitemap.xml.tsx
+import { getServerSideSitemapLegacy } from 'next-sitemap'
+import { MakeswiftPageResult } from '@makeswift/runtime/next'
+import { client } from '@makeswift/client'
+
+const DOMAIN = 'https://example.com'
+const DEFAULT_PRIORITY = 0.75
+const DEFAULT_FREQUENCY = 'hourly'
+
+function pageToSitemapItem(page: MakeswiftPageResult) {
+  const pageUrl = new URL(page.path, DOMAIN)
+  return {
+    loc: pageUrl.href,
+    lastmod: page.createdAt,
+    changefreq: page.sitemapFrequency ?? DEFAULT_FREQUENCY,
+    priority: page.sitemapPriority ?? DEFAULT_PRIORITY,
+    alternateRefs: page.localizedVariants.map(variant => {
+      const localizedPath = `/${variant.locale}/${variant.path}`
+      const localizedPageUrl = new URL(localizedPath, DOMAIN)
+      return {
+        hreflang: variant.locale,
+        href: localizedPageUrl.href,
+      }
+    }),
+  }
+}
+
+export async function getServerSideProps(context) {
+  const sitemap = client
+    .getPages()
+    .filter(page => !page.excludedFromSearch)
+    .map(page => pageToSitemapItem(page))
+    .toArray()
+
+  return getServerSideSitemapLegacy(context, sitemap)
+}
+
+export default function Sitemap() {}
+
+```
+
+BREAKING: The exported `MakeswiftPage` type has been replaced with
+`MakeswiftPageResult`, which includes several more data fields from the
+Makeswift page.
```

Comment:
Do we need to rename it? `MakeswiftPageResult` is kind of an awkward name.

Reply:
Do you have any suggestions? I wanted to be more specific than `Page` or `MakeswiftPage` to avoid collision with the component of the same/similar name that renders page snapshots.

Reply:
I'd just keep it as `MakeswiftPage`, since it's an established name we have and it fits with `MakeswiftPageDocument` and `MakeswiftPageSnapshot`. I think if we want to revise our naming convention/namespacing with regards to API types vs components, we need to do it wholesale.

Reply:
I agree on doing this wholesale at another time and keeping it as `MakeswiftPage` for now